### PR TITLE
[SMALLFIX] Fix an issue in getFileStatus and add unit test

### PR DESCRIPTION
**Pull request by apc999**

### CODE REVIEW

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java [hunk 1]
```diff
@@ -225,4 +249,44 @@ private void mockUserGroupInformation() throws IOException {
...
+   * @param infos the list of file infos
+   */
+  private void mockAlluxioFsListStatusHelper(Path path, FileInfo... infos) throws Exception {
+    List<URIStatus> uris = new ArrayList<>();
+    for (FileInfo info : infos) {
+      uris.add(new URIStatus(info));
+    }
+    alluxio.client.file.FileSystem alluxioFs =
+        Mockito.mock(alluxio.client.file.FileSystem.class);
+    Whitebox.setInternalState(FileSystem.class, "sFileSystem", alluxioFs);
```
Comment (calvinjia) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: Is it possible to avoid the use of whitebox?

Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: Could we change AbstractFileSystem to take an optional FileSystem in its constructor, and mock it that way?

Comment (apc999) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: I think that will work

Comment (apc999) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: @aaudiber actually that will not work. The reason is the instance `alluxio.hadoop.FileSystem` is created from Hadoop code by reflection with no constructor:
```java
URI uri = URI.create(Constants.HEADER + "localhost:19998/");
    org.apache.hadoop.fs.FileSystem alluxioHadoopFs =
        org.apache.hadoop.fs.FileSystem.get(uri, getConf());
```
We could directly create an `alluxio.hadoop.FileSystem` instance rather from the Hadoop factory code, but that will cover less.

Comment (aaudiber) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: @apc999 Could we call the constructor directly instead of using reflection?

**Comment by apc999** [comment]
@calvinjia @aaudiber  Could you take another look?

File: core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java (hunk 1 above)
Comment (apc999) [inline core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java]: added

File: core/client/src/main/java/alluxio/hadoop/FaultTolerantFileSystem.java [hunk 2]
```diff
@@ -36,6 +36,15 @@ public FaultTolerantFileSystem() {
     super();
   }
 
+  /**
+   * Constructs a new
```
Comment (aaudiber) [inline core/client/src/main/java/alluxio/hadoop/FaultTolerantFileSystem.java]: nit: line wraps too early

File: core/client/src/main/java/alluxio/hadoop/FaultTolerantFileSystem.java [hunk 3]
```diff
@@ -36,6 +36,15 @@ public FaultTolerantFileSystem() {
...
+   * {@link FaultTolerantFileSystem} instance with specified a {@link alluxio.client.file.FileSystem}
```
Comment (apc999) [inline core/client/src/main/java/alluxio/hadoop/FaultTolerantFileSystem.java]: done

File: core/client/src/main/java/alluxio/hadoop/FaultTolerantFileSystem.java (hunk 2 above)
Comment (apc999) [inline core/client/src/main/java/alluxio/hadoop/FaultTolerantFileSystem.java]: fixed

File: core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java [hunk 4]
```diff
@@ -91,6 +91,7 @@
    */
   AbstractFileSystem(FileSystem fileSystem) {
     sFileSystem = fileSystem;
+    sInitialized = true;
```
Comment (aaudiber) [inline core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: is there any reason for this class to store the filesystem statically? Could we make the filesystem field non-static and use `FileSystem.Factory.get()` in the no-arg constructor, e.g.

```java
AbstractFileSystem() {
  this(FileSystem.Factory.get());
}
```

Comment (apc999) [inline core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: `FileSystem.Factory.get()` must be called after `Configuration` initialized (in here `Configuration` may be initialized during `intialize()` methid. So we could not set the `FileSystem` member variable in construction time but lazily.

**Comment by apc999** [comment]
@aaudiber I played around with the idea that making `sFileSystem` a member variable from static, but it requires more change in other unit tests as we are using whiteboxing to reset client context. I think this class `AbstractFileSystem` is old enough and we should really cleanup it. Meanwhile, this is a bug-fix PR and I don't want to introduce more refactoring to enlarge the footprint. therefore I made a "hacky" change here

**Comment by calvinjia** [comment]
LGTM, based on how difficult it is to write this unit test, we need to reconsider the implementation of this class.

**Pull request merged**